pkg/parser/queryfilterextractor/clickhouse_test.go

```diff
@@ -0,0 +1,954 @@
+		    greatest(curr.success - ifNull(prev.success, 0), 0) AS d_success,
+		    greatest(curr.desired - ifNull(prev.desired, 0), 0) AS d_desired
+		  FROM flags AS curr
+		  LEFT JOIN flags AS prev
+		    ON curr.job = prev.job
+		   AND prev.ts  = curr.ts - toIntervalSecond(bucket_s)
+		), job_first_success AS (
+		  SELECT job, min(ts) AS first_success_ts
+		  FROM flags
+		  WHERE complete_now
+		  GROUP BY job
+		)
+		SELECT  d.ts,
+		  sum(
+		    multiIf(
+		      greatest(d_failed + d_success - d_desired, 0) > 0      AND (
+		        js.first_success_ts IS NULL
+		        OR js.first_success_ts < d.ts
+		        ),
+		      greatest(d_failed + d_success - d_desired, 0),
+		      0    )
+		  ) AS value FROM deltas AS d
+		LEFT JOIN job_first_success AS js
+		  ON d.job = js.job
+		GROUP BY d.ts
+		ORDER BY d.ts;
+					`,
+			wantMetrics: []string{"k8s.job.failed_pods", "k8s.job.successful_pods", "k8s.job.desired_successful_pods"},
+			wantGroupBy: []string{"ts"},
+		},
+		{
+			name: "TC6 - Outer GROUP BY with ClickHouse dialect (backticks)",
+			query: `
+		SELECT
+		    ` + "`os.type`" + `,
+		    state,
+		    ` + "`host_name`" + `,
+		    ts,
+		    max(per_series_value) AS value FROM (
+		    SELECT
+		        fingerprint,
+		        any(` + "`os.type`" + `) AS ` + "`os.type`" + `,
+		        any(state) AS state,
+		        any(` + "`host_name`" + `) AS ` + "`host_name`" + `,
+		        toStartOfInterval(toDateTime(intDiv(unix_milli, 1000)), INTERVAL 60 SECOND) AS ts,
+		        max(value) AS per_series_value
+		    FROM signoz_metrics.distributed_samples_v4
+		    INNER JOIN (
+		        SELECT DISTINCT
+		            JSONExtractString(labels, 'os.type') AS ` + "`os.type`" + `,
+		            JSONExtractString(labels, 'state') AS state,
+		            JSONExtractString(labels, 'host_name') AS ` + "`host_name`" + `,
+		            fingerprint
+		        FROM signoz_metrics.time_series_v4_1day
+		        WHERE
+		            metric_name IN ['system.memory.usage']
+		            AND temporality = 'Unspecified'            AND __normalized = false            AND unix_milli >= 1650931200000            AND unix_milli < 1651078380000            AND JSONExtractString(labels, 'host.name') = 'signoz-host'    ) AS filtered_time_series
+		    USING fingerprint
+		    WHERE
+		        metric_name IN ['system.memory.usage']
+		        AND unix_milli >= 1650991980000        AND unix_milli < 1651078380000        AND bitAnd(flags, 1) = 0    GROUP BY
+		        fingerprint,
+		        ts
+		    ORDER BY
+		        fingerprint,
+		        ts
+		)
+		WHERE isNaN(per_series_value) = 0
+		GROUP BY
+		    ` + "`os.type`" + `,
+		    state,
+		    ` + "`host_name`" + `,
+		    ts
+		ORDER BY
+		    ` + "`os.type`" + ` DESC,
+		    state ASC,
+		    ` + "`host_name`" + ` ASC,
+		    ts ASC
+					`,
+			wantMetrics: []string{"system.memory.usage"},
+			wantGroupBy: []string{"os.type", "state", "host_name", "ts"},
```

Comment:
This is correct, but to be able to exclude, we should know the original attributes right? otherwise, we don't know when the corresponding group initially ingested. For example, here `host_name` is the alias and the result label, but the lack of context about the original attribute `host.name` means we can't search for the metadata with the correct attribute name.